### PR TITLE
fix: time input on drop keyboard direction

### LINKED ISSUE #8124: TimeInput - up/down arrow key behavior incorrect within drop
When navigating through the list of items within the drop pressing the up arrow moves focus down within the list and pressing down arrow moves focus up. This should be reversed so that up arrow moves up and down arrow moves down.

### PR DESCRIPTION
**Pull request by mushroomgenie**
#### What does this PR do?

This PR fixes the arrow key direction in TimeInput on the drop component.

#### Where should the reviewer start?

The reviewer can start by looking at `src/js/components/TimeInput/TimeInputPopup.js` and the tests.

#### What testing has been done on this PR?

Three new tests have been added to check the validity of the change.

#### Do Jest tests follow these best practices?

- [x] `screen` is used for querying.
- [x] The correct query is used. (Refer to [this list of queries](https://testing-library.com/docs/queries/about/#priority))
- [ ] `asFragment()` is used for snapshot testing.

#### What are the relevant issues?

Fixes #8124 

#### Screenshots (if appropriate)
time-input-key-up-down

### CODE REVIEW

**Review by Sulaymon333 — APPROVED** [review]
Looks good

File: src/js/components/TimeInput/__tests__/TimeInput-test.tsx [hunk 1]
```diff
@@ -2190,4 +2190,68 @@ describe('TimeInput', () => {
...
+  test('initialises hour with zero on ArrowUp in hour inside drop', async () => {
+    const user = userEvent.setup();
+
+    render(
+      <Grommet>
+        <TimeInput format="24" />
+      </Grommet>,
+    );
+
+    const choostTimeButton = screen.getByRole('button', {
```
Comment (britt6612) [inline src/js/components/TimeInput/__tests__/TimeInput-test.tsx]: think typo 

`chooseTimeButton`

File: src/js/components/TimeInput/useSectionedTimeField.js [hunk 2]
```diff
@@ -241,7 +241,11 @@ export const useSectionedTimeField = ({
       const current = sections[key];
 
       if (current === undefined) {
-        setSectionValue(section, delta > 0 ? minValue : maxValue);
+        if (section === SECTION_HOUR) {
+          setSectionValue(section, format === '12' ? maxValue : minValue);
+        } else {
+          setSectionValue(section, delta > 0 ? minValue : maxValue);
```
[the remark this answers is not in the document]
Comment (britt6612) [reply src/js/components/TimeInput/useSectionedTimeField.js]: Could we preserve the existing directional initialization for the closed input?

- 24-hour ArrowUp → 00
- 24-hour ArrowDown → 23
- 12-hour ArrowUp → 01
- 12-hour ArrowDown → 12

right now with this PR this is happening 

In 24-hour format, an empty hour now initializes to 00 for either arrow direction.
In 12-hour format, an empty hour now initializes to 12 for either arrow direction.

@MikeKingdom @halocline 
would you 2 agree with the existing behvaior?

[the remark this answers is not in the document]
Comment (MikeKingdom) [reply src/js/components/TimeInput/useSectionedTimeField.js]: @britt6612 What we originally settled on was what you're requesting (not the 00/12 for either arrow direction)

[the remark this answers is not in the document]
Comment (britt6612) [reply src/js/components/TimeInput/useSectionedTimeField.js]: @mushroomgenie 

With the behavior initializes an empty hour to the format’s default for both ArrowUp and ArrowDown, could we apply the same rule to the minute and second sections?

When the minute or second section is empty:

ArrowUp should initialize it to 00
ArrowDown should also initialize it to 00

This would make the initial empty-section behavior consistent across hours, minutes, and seconds. Subsequent arrow presses should continue incrementing or decrementing normally.

[the remark this answers is not in the document]
Comment (mushroomgenie) [reply src/js/components/TimeInput/useSectionedTimeField.js]: @britt6612 Sure I'll add that.

[the remark this answers is not in the document]
Comment (mushroomgenie) [reply src/js/components/TimeInput/useSectionedTimeField.js]: @britt6612 Just one question, what should the hour initialise to on the 12 hour format ? 12 or 01 ?

[the remark this answers is not in the document]
Comment (MikeKingdom) [reply src/js/components/TimeInput/useSectionedTimeField.js]: > @britt6612 Just one question, what should the hour initialise to on the 12 hour format ? 12 or 01 ?

We've gone back and forth on that but I think 01 may make more sense and consistent with some other time input components

[the remark this answers is not in the document]
Comment (britt6612) [reply src/js/components/TimeInput/useSectionedTimeField.js]: I'd agree with @MikeKingdom  and having the 01

[the remark this answers is not in the document]
Comment (mushroomgenie) [reply src/js/components/TimeInput/useSectionedTimeField.js]: @MikeKingdom @britt6612 made the changes

File: src/js/components/TimeInput/__tests__/TimeInput-test.tsx (hunk 1 above)
Comment (mushroomgenie) [reply src/js/components/TimeInput/__tests__/TimeInput-test.tsx]: @britt6612 fixed this.

File: src/js/components/TimeInput/useSectionedTimeField.js [hunk 3]
```diff
@@ -241,7 +241,12 @@ export const useSectionedTimeField = ({
       const current = sections[key];
 
       if (current === undefined) {
-        setSectionValue(section, delta > 0 ? minValue : maxValue);
+        setSectionValue(section, minValue);
+        // if (section === SECTION_HOUR) {
```
Comment (britt6612) [inline src/js/components/TimeInput/useSectionedTimeField.js]: @mushroomgenie can we remove?

File: src/js/components/TimeInput/__tests__/TimeInput-test.tsx (hunk 1 above)
Comment (britt6612) [reply src/js/components/TimeInput/__tests__/TimeInput-test.tsx]: I do not see if fixed on my end

File: src/js/components/TimeInput/__tests__/TimeInput-test.tsx [hunk 4]
```diff
@@ -2193,4 +2190,68 @@ describe('TimeInput', () => {
...
+    await user.click(chooseTimeButton);
+    expect(getDisplayInput()).toHaveValue('hh:mm');
+
+    await user.keyboard('{ArrowDown}');
+    expect(getDisplayInput()).toHaveValue('00:mm');
+    await user.keyboard('{ArrowDown}');
+    expect(getDisplayInput()).toHaveValue('01:mm');
+  });
+
+  test('initialises hour with zero on ArrowUp in hour inside drop', async () => {
```
Comment (britt6612) [inline src/js/components/TimeInput/__tests__/TimeInput-test.tsx:2215]: Any chance we could add an explicit test for an empty seconds section, so the new 00 initialization behavior is covered for both minutes and seconds?

File: src/js/components/TimeInput/__tests__/TimeInput-test.tsx (hunk 1 above)
Comment (Sulaymon333) [reply src/js/components/TimeInput/__tests__/TimeInput-test.tsx]: @britt6612 fixed the typo via this commit [4621b31](https://github.com/grommet/grommet/pull/8125/commits/4621b31334e268d66b93370d0bc0ccc7e7995b10)

cc: @mushroomgenie

File: src/js/components/TimeInput/useSectionedTimeField.js (hunk 3 above)
Comment (mushroomgenie) [reply src/js/components/TimeInput/useSectionedTimeField.js]: @britt6612 removed the comments.

File: src/js/components/TimeInput/__tests__/TimeInput-test.tsx (hunk 4 above)
Comment (mushroomgenie) [reply src/js/components/TimeInput/__tests__/TimeInput-test.tsx:2215]: Sure I'll add that.

Comment (britt6612) [reply src/js/components/TimeInput/__tests__/TimeInput-test.tsx:2215]: thanks! let me know if you have any issues

File: src/js/components/TimeInput/useSectionedTimeField.js (hunk 3 above)
Comment (britt6612) [reply src/js/components/TimeInput/useSectionedTimeField.js]: thank you

File: src/js/components/TimeInput/__tests__/TimeInput-test.tsx (hunk 4 above)
Comment (mushroomgenie) [reply src/js/components/TimeInput/__tests__/TimeInput-test.tsx:2215]: @britt6612 added the test.

**Comment by britt6612** [comment]
@mushroomgenie 

> Review effort: Lite
> Findings: 1 Medium severity
> 
> Open (1)
> Medium severity [incrementSection is also used by the non-drop keyboard handler (TimeInput.js:541-553). This…](https://github.com/grommet/grommet/pull/8125#discussion_r4032099838)

think co pilots comment is valid The shared `incrementSection` uninitialized-value change affects the non-drop path too. So when I focus on the TimeInput and do not open the drop and I arrow-down I get "00" where before I got "23"

you can see the behavior here comparing these two examples the first one is the preview link from this PR change, second link is todays storybook build from whats currently on master.

1. https://deploy-preview-8125--sad-tereshkova-173d07.netlify.app/?path=/story/input-timeinput-simple--simple&globals=theme:hpe
2. https://storybook.grommet.io/?path=/story/input-timeinput-simple--simple&globals=theme:hpe

**Comment by mushroomgenie** [comment]
> @mushroomgenie
> 
> > Review effort: Lite
> > Findings: 1 Medium severity
> > Open (1)
> > Medium severity [incrementSection is also used by the non-drop keyboard handler (TimeInput.js:541-553). This…](https://github.com/grommet/grommet/pull/8125#discussion_r4032099838)
> 
> think co pilots comment is valid The shared `incrementSection` uninitialized-value change affects the non-drop path too. So when I focus on the TimeInput and do not open the drop and I arrow-down I get "00" where before I got "23"
> 
> you can see the behavior here comparing these two examples the first one is the preview link from this PR change, second link is todays storybook build from whats currently on master.
> 
> 1. https://deploy-preview-8125--sad-tereshkova-173d07.netlify.app/?path=/story/input-timeinput-simple--simple&globals=theme:hpe
> 2. https://storybook.grommet.io/?path=/story/input-timeinput-simple--simple&globals=theme:hpe

@britt6612 Makes sense, how about we add an optional isDrop boolean parameter to the incrementSection function, and retain the previous behaviour and pass isDrop only when the drop is open.

**Comment by britt6612** [comment]
> > @mushroomgenie
> > > Review effort: Lite
> > > Findings: 1 Medium severity
> > > Open (1)
> > > Medium severity [incrementSection is also used by the non-drop keyboard handler (TimeInput.js:541-553). This…](https://github.com/grommet/grommet/pull/8125#discussion_r4032099838)
> > 
> > 
> > think co pilots comment is valid The shared `incrementSection` uninitialized-value change affects the non-drop path too. So when I focus on the TimeInput and do not open the drop and I arrow-down I get "00" where before I got "23"
> > you can see the behavior here comparing these two examples the first one is the preview link from this PR change, second link is todays storybook build from whats currently on master.
> > 
> > 1. https://deploy-preview-8125--sad-tereshkova-173d07.netlify.app/?path=/story/input-timeinput-simple--simple&globals=theme:hpe
> > 2. https://storybook.grommet.io/?path=/story/input-timeinput-simple--simple&globals=theme:hpe
> 
> @britt6612 Makes sense, how about we add an optional isDrop boolean parameter to the incrementSection function, and retain the previous behaviour and pass isDrop only when the drop is open.

that should work!

**Comment by mushroomgenie** [comment]
> > > @mushroomgenie
> > > > Review effort: Lite
> > > > Findings: 1 Medium severity
> > > > Open (1)
> > > > Medium severity [incrementSection is also used by the non-drop keyboard handler (TimeInput.js:541-553). This…](https://github.com/grommet/grommet/pull/8125#discussion_r4032099838)
> > > 
> > > 
> > > think co pilots comment is valid The shared `incrementSection` uninitialized-value change affects the non-drop path too. So when I focus on the TimeInput and do not open the drop and I arrow-down I get "00" where before I got "23"
> > > you can see the behavior here comparing these two examples the first one is the preview link from this PR change, second link is todays storybook build from whats currently on master.
> > > 
> > > 1. https://deploy-preview-8125--sad-tereshkova-173d07.netlify.app/?path=/story/input-timeinput-simple--simple&globals=theme:hpe
> > > 2. https://storybook.grommet.io/?path=/story/input-timeinput-simple--simple&globals=theme:hpe
> > 
> > 
> > @britt6612 Makes sense, how about we add an optional isDrop boolean parameter to the incrementSection function, and retain the previous behaviour and pass isDrop only when the drop is open.
> 
> that should work!

@britt6612 added the change. FYI an e2e test always fails on the first attempt, always have to push an empty commit to get it to pass. Is this a known issue ?

**Comment by britt6612** [comment]
> > > > @mushroomgenie
> > > > > Review effort: Lite
> > > > > Findings: 1 Medium severity
> > > > > Open (1)
> > > > > Medium severity [incrementSection is also used by the non-drop keyboard handler (TimeInput.js:541-553). This…](https://github.com/grommet/grommet/pull/8125#discussion_r4032099838)
> > > > 
> > > > 
> > > > think co pilots comment is valid The shared `incrementSection` uninitialized-value change affects the non-drop path too. So when I focus on the TimeInput and do not open the drop and I arrow-down I get "00" where before I got "23"
> > > > you can see the behavior here comparing these two examples the first one is the preview link from this PR change, second link is todays storybook build from whats currently on master.
> > > > 
> > > > 1. https://deploy-preview-8125--sad-tereshkova-173d07.netlify.app/?path=/story/input-timeinput-simple--simple&globals=theme:hpe
> > > > 2. https://storybook.grommet.io/?path=/story/input-timeinput-simple--simple&globals=theme:hpe
> > > 
> > > 
> > > @britt6612 Makes sense, how about we add an optional isDrop boolean parameter to the incrementSection function, and retain the previous behaviour and pass isDrop only when the drop is open.
> > 
> > 
> > that should work!
> 
> @britt6612 added the change. FYI an e2e test always fails on the first attempt, always have to push an empty commit to get it to pass. Is this a known issue ?

@mushroomgenie yes the e2e has some issues its not related to your PR! Thank you so much for you work and follow through I will look at this now!

**Review by britt6612 — APPROVED** [review]

**Review by MikeKingdom — APPROVED** [review]
Looks good although it needs a couple changes from master like the clock button change.

**Review by Sulaymon333 — APPROVED** [review]
LGTM. Thank you for working on this @mushroomgenie

**Review by jcfilben — APPROVED** [review]
